resources/camino-creator/stores/useCreatorStore.ts

```diff
@@ -84,6 +84,8 @@ export const useCreatorStore = defineStore("creator", () => {
       computed(() =>
         selectors.selectNextTourStopStartPoint(state, tourId, stopId)
       ),
+    findFirstValuedTargetPoint: (tourId: number) =>
+      computed(() => selectors.findFirstValuedTargetPoint(state, tourId)),
```

Comment:
At some point, can you explain to me why these have to be wrapped in a computed()? It seems like they're called explicitly, not reactively?

Reply:
In pinia, [getters are equivalent to computed](https://pinia.vuejs.org/core-concepts/getters.html) under the hood.  By making getters computed, they can trigger a component re-render when the state changes. Also, computed they are [cached](https://vuejs.org/guide/essentials/computed.html#computed-caching-vs-methods) based on their reactive dependencies, which should help with some of the expensive array ops.

You're right: often, we're calling these just like methods and the value is immediately used, so they don't "have to" be wrapped. But, I'm keeping the same computed pattern and  hoping the caching helps if a different component uses the same value.

Pinia does a little magic wrapping and unwrapping of computed values, and normally hides this computed stuff. I found the magic nice in the simple cases when values are purely derived from state, but difficult to understand when we need to pass arguments (e.g. a tour or stop id).

At some point, I decided to leave the magic behind and follow the function pattern described in [this Issue](https://github.com/vuejs/pinia/issues/978). I'm MUCH happier for it. It looks a little strange to add a `.value` at the end of a getter call, but at least it makes sense now in my mind: "Oh – that's because it's returning a computed ref and not the actual value"

At least that's my understanding... does that help clarify?

Reply:
yep, thanks!